src/main/java/com/redhat/devtools/lsp4ij/LanguageServersRegistry.java

```diff
@@ -84,12 +82,8 @@ private void loadServersAndMappingsFromExtensionPoint() {
             Language language = Language.findLanguageByID(extension.language);
             if (language != null) {
                 String serverId = extension.serverId;
-                List<ServerMapping> mappingsForServer = mappings.get(serverId);
-                if (mappingsForServer == null) {
-                    mappingsForServer = new ArrayList<>();
-                    mappings.put(serverId, mappingsForServer);
-                }
-                @NotNull String languageId = StringUtils.isEmpty(extension.languageId) ? language.getID() : extension.languageId;
```

Comment:
why not keep that? (with `language.getID().toLowerCase()`)

Reply:
The languageId is ûsed when didOpen notification is sent.

The compute of this languageId tries to get it with the languagr file and if it is not found it uses the base language and retry it.

In other words mapping should store the spécified languageId and null otherwise to return the specified languageId and not by the ij kanguage.getID by default